File: 2736-minimum-additions-to-make-valid-string/minimum-additions-to-make-valid-string.c
int addMinimum(char * word){
    int a=0,i=0;
    int len=strlen(word);
    char expected='a';
    while (i < len) {
        if (word[i] == expected)
        i++;
        else
        a++;        
        expected++;
        if (expected > 'c') 
        expected = 'a';
    }
    while (expected != 'a') {
        a++;
        expected++;
        if (expected > 'c') 
        expected = 'a';
    }
    
    return a;
}

File: 2736-minimum-additions-to-make-valid-string/minimum-additions-to-make-valid-string.java
class Solution {
    public int addMinimum(String word) {
        int additions = 0;
        int i = 0;
        char expected = 'a';
        
        while (i < word.length()) {
            if (word.charAt(i) == expected) {
                i++;
            } else {
                additions++;
            }
            
            expected++;
            if (expected > 'c') expected = 'a';
        }
        
        // Add remaining characters to complete the sequence
        while (expected != 'a') {
            additions++;
            expected++;
            if (expected > 'c') expected = 'a';
        }
        
        return additions;
    }
}